### DISCUSSION TITLE
Update Template CI action fails to push changes

### DISCUSSION THREAD

**Discussion by mig5** [upptime/upptime#743] [Q&A]
My 'Update Template CI' action seems to fail to update the template. The 'Update template' action step shows this:

```
To https://github.com/XXX/upptime
 ! [remote rejected] master -> master (refusing to allow a GitHub App to create or update workflow `.github/workflows/graphs.yml` without `workflows` permission)
error: failed to push some refs to 'https://github.com/XXX/upptime'
```

I have checked and the `GH_PAT` Action secret I added years ago, is still present. I checked the token's permissions itself and it definitely has the 'workflow' permission.

What am I doing wrong?

**Comment by mig5** [comment]
Think I figured it out

Back in https://github.com/upptime/upptime/commit/71d15e55e39f3e5bd6fe4b7051ebbc6a7aa676ad you changed it to GITHUB_TOKEN. 

My upptime probably pulled down this change.

This change didn't work since GH_PAT was set

You fixed it in https://github.com/upptime/upptime/commit/71d15e55e39f3e5bd6fe4b7051ebbc6a7aa676ad but of course, a catch-22, because the incorrect secret name/value being used, I could not pull this update (or any future updates). Ouch.

I have manually updated my workflow files, but of course, Github Actions isn't working right now.. we'll see if it works later.

**Reply by AnandChowdhary** [reply]
Can you confirm that your GH_PAT personal access token has the "workflow" scope set up?

**Reply by mig5** [reply]
Yes it does.

The problem as I see it is that the code expects `GITHUB_TOKEN` and not `GH_PAT`, though.

Is there a way I can manually update the template without using Actions?

![Screenshot_2023-03-08_07-20-17](https://user-images.githubusercontent.com/99173/223543205-3c6a642c-5f94-4d48-8408-6b9c19de0c5b.png)

**Reply by AnandChowdhary** [reply]
No, if I'm not mistaken, the most recent updates allows you to use Upptime without a PAT, but using a PAT is still supported: https://github.com/upptime/upptime/blob/master/.github/workflows/uptime.yml#L26. We only fallback to using the GITHUB_TOKEN if a PAT is unavailable.

**Reply by mig5** [reply]
I think what you're not understanding is that I am *not* on the newest version because the very Action that runs 'Update Template' is failing *because* it expects `GITHUB_TOKEN`.

I do not believe the version I am on, uses GH_PAT with a fallback to GITHUB_TOKEN. It's somehow on a version that expects GITHUB_TOKEN only, at least that is my theory. Do you see the catch 22?

How can I show you what version I am using?

**Reply by AnandChowdhary** [reply]
Aah I understand now. What if you manually delete the .github/workflows directory in your repository and instead copy and paste each file from the upstream upptime/upptime repository?

**Reply by mig5** [reply]
I just tried that, then I ran the 'Update Template CI' action, and I still seem  to get this strange error:

```
Fetching the repository
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +80f1ab5a90d72672e7e33ede0545f5cb6d8ee52d:refs/remotes/origin/master
  Error: fatal: could not read Username for 'https://github.com/': terminal prompts disabled
  The process '/usr/bin/git' failed with exit code 128
  Waiting 16 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +80f1ab5a90d72672e7e33ede0545f5cb6d8ee52d:refs/remotes/origin/master
  Error: fatal: could not read Username for 'https://github.com/': terminal prompts disabled
  The process '/usr/bin/git' failed with exit code 128
  Waiting 15 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +80f1ab5a90d72672e7e33ede0545f5cb6d8ee52d:refs/remotes/origin/master
  Error: fatal: could not read Username for 'https://github.com/': terminal prompts disabled
  Error: The process '/usr/bin/git' failed with exit code 128
  ```

**Comment by mig5** [comment]
Actions are working again, but it didn't like me changing GITHUB_TOKEN to GH_PAT. I get prompts for a username:

```
Fetching the repository
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +c67b92803e4ceafea00d4821d662dea2cc7ee598:refs/remotes/origin/master
  Error: fatal: could not read Username for 'https://github.com/': terminal prompts disabled
  The process '/usr/bin/git' failed with exit code 128
  Waiting 12 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +c67b92803e4ceafea00d4821d662dea2cc7ee598:refs/remotes/origin/master
  Error: fatal: could not read Username for 'https://github.com/': terminal prompts disabled
  The process '/usr/bin/git' failed with exit code 128
  Waiting 17 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +c67b92803e4ceafea00d4821d662dea2cc7ee598:refs/remotes/origin/master
  Error: fatal: could not read Username for 'https://github.com/': terminal prompts disabled
  Error: The process '/usr/bin/git' failed with exit code 128
  ```
  
  What is going wrong? How do I get onto the latest version, is there something I can do manually with my upptime repo?
